pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -173,11 +173,17 @@ func reconcileObjectMeta(existing *metav1.ObjectMeta, desired metav1.ObjectMeta)
 }
 
 func (r *Reconciler) reconcileService(logger *zap.SugaredLogger, el *v1alpha1.EventListener) error {
+	// for backward compatibility with original behavior
+	var serviceType = el.Spec.ServiceType
+	if el.Spec.Resources.KubernetesResource != nil && el.Spec.Resources.KubernetesResource.ServiceType != "" {
+		serviceType = el.Spec.Resources.KubernetesResource.ServiceType
```

Comment:
Do we also want to add some defaulting logic in el.SetDefaults to change el.Spec.ServiceType to el.Spec.Resources.KubernetesResource.ServiceType?

Reply:
I think we don't require because even if `ServiceType` is empty k8s itself set `ClusterIP` as default

Reply:
Oh, what I meant by defaulting is something like

```
  if el.Spec.ServiceType != nil {
    el.Spec.Resources.KubernetesResource.ServiceType = el.Spec.ServiceType
```

This is so that we can deprecate/remove the `el.Spec.ServiceType` field later.